src/api/entities/Identity/index.ts

```diff
@@ -981,4 +986,64 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
 
     return boolToBoolean(rawIsApproved);
   }
+
+  /**
+   * Returns the list of MultiSig accounts linked with this Identity along with the signatories for whom the MultiSig acts as the signer
+   *
+   * @note this query can be potentially **SLOW** depending on the number of MultiSigs present on the chain
+   */
+  public async getMultiSigSigners(): Promise<MultiSigSigner[]> {
+    const {
+      context,
+      context: {
+        polymeshApi: {
+          query: {
+            multiSig: { multiSigToIdentity, multiSigSigners },
+          },
+        },
+      },
+      did,
+    } = this;
+
+    const entries = await multiSigToIdentity.entries();
+
+    const multiSigs: Record<string, Signer[]> = {};
+
+    const rawSigners = await Promise.all(
+      entries
+        .filter(([, rawIdentityId]) => {
+          const identity = identityIdToString(rawIdentityId);
+          return identity === did;
+        })
+        .map(
+          ([
+            {
+              args: [rawMultiSigAccount],
+            },
+          ]) => {
+            multiSigs[accountIdToString(rawMultiSigAccount)] = [];
+            return multiSigSigners.entries(rawMultiSigAccount);
+          }
+        )
+    );
+
+    rawSigners.forEach(rawSigner => {
+      rawSigner.forEach(
+        ([
+          {
+            args: [rawMultiSigAccount, signatory],
+          },
+        ]) => {
+          multiSigs[accountIdToString(rawMultiSigAccount)].push(
```

Comment:
```suggestion
          const accountId = accountIdToString(rawMultiSigAccount)
          multiSigs[accountId].push(
```

(nit) I think its more readable having the key on its own line.